pkg/kube/kube.go

```diff
@@ -55,6 +59,24 @@ func GetClient() (*kubernetes.Clientset, error) {
 	return kubernetes.NewForConfig(config)
 }
 
+func CreateSingleClient() (client.Client, error) {
```

Comment:
Why did you opt to create a custom function rather than using the clientset which is already provided?

Reply:
After an offline discussion we decided to opt for a special secret reading function rather than a custom function to make a client